### PR TITLE
session-note cleanup (via ChatGPT generated code)

### PR DESCRIPTION
**Pull request by MichaelRWolf**
For all session notes, performed these cleanup steps
- Add missing "Session Date" header
- Removed In-Active Co-Authors
- Normalized "Co-Author" headings by removing spurious text (i.e. for today), leaving just "# Co-Authors" as simple header

For template file, created "How to" section as reminder for how create a session file from the template file.

P.S.  I started this project by hand on today's notes.  Then kept noticing more little problems.  I was about ready to 'git commit' when something bad happened.  So, after a swim, I started a Perl script to automate my fat-finger approach.  

Then I asked ChatGPT to get me started.  

Then I had it switch to Python mid-session.  

Then it wrote ALL the code to do the cleanup.  Yes, 100.000% of the code is generated from my (blunt tool called) English.

The prompt and the Python script are in the repo.

There's also a quick/dirty summary of before & after cleanup for counts of co-authors.

### CODE REVIEW

File: session-notes/session-notes-2023-04-27.md [hunk 1]
```diff
@@ -1,23 +1,12 @@
...
+## Co-Authors
 
 Co-Authored-By: Gregor Riegler <rieglerg85@gmail.com>
 Co-Authored-By: Joel Silberman <42779942+jcs-instructor@users.noreply.github.com>
 Co-Authored-By: Nitsan Avni <nitsanav@gmail.com>
 Co-Authored-By: Michael R. Wolf <MichaelRWolf@att.net>
 
-## Active Co-Authors
-
-## Inactive Co-Authors
```
Comment (nitsanavni) [inline session-notes/session-notes-2023-04-27.md:10]: 🔥

File: session-notes/session-notes-2023-04-27.md [hunk 2]
```diff
@@ -1,23 +1,12 @@
...
-Co-Authored-By: arrockt <cariari385@gmail.com>
-Co-Authored-By: Austin Chadwick <austin.chadwick11@gmail.com>
-Co-Authored-By: Blessed538 <blesseddominic98@gmail.com>
-Co-Authored-By: Eddie Bush <eddie@craftsmanshipcounts.com>
-Co-Authored-By: Gabriel Mbaiorga <gabrielmbaiorga@gmail.com>
-Co-Authored-By: Idan Melamed <idanmel@gmail.com>
-Co-Authored-By: Rea <reasu@protonmail.com>
-Co-Authored-By: Tsvetan Tsvetanov <cpi.cecko@gmail.com>
-Co-Authored-By: Willem Larsen <willemlarsen@gmail.com>
+## Co-Authors
```
Comment (nitsanavni) [inline session-notes/session-notes-2023-04-27.md:9]: dup

File: session-notes/session-notes-2023-05-04.md [hunk 3]
```diff
@@ -1,24 +1,14 @@
-# Co-Authors (This Session)
+# Session Date: 2023-05-04
+## Co-Authors
 
-## Active Co-Authors
```
Comment (nitsanavni) [inline session-notes/session-notes-2023-05-04.md:3]: ❤️

File: session-notes/cleanup-notes/cleanup_session_notes.prompt [hunk 4]
```diff
@@ -0,0 +1,82 @@
+Ignore previous instructions.
+
+# Task: perform cleanup on files
```
Comment (nitsanavni) [inline session-notes/cleanup-notes/cleanup_session_notes.prompt:3]:
```suggestion
# Task: write a script that'll perform cleanup on files
```

File: session-notes/cleanup-notes/cleanup_session_notes.prompt [hunk 5]
```diff
@@ -0,0 +1,82 @@
+Ignore previous instructions.
```
Comment (nitsanavni) [inline session-notes/cleanup-notes/cleanup_session_notes.prompt:1]: Why?

File: session-notes/cleanup-notes/doit [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+cd  "${note_directory}" || exit 2
+
+
+
+for arg in "$@"
+do
+    if [[ ! -f "$arg" ]]; then
+	date="$arg"
+	filename="session-notes-${date}.md"
```
Comment (nitsanavni) [inline session-notes/cleanup-notes/doit:16]: Could've done this inside the perl/python script

File: session-notes/cleanup-notes/doit [hunk 7]
```diff
@@ -0,0 +1,25 @@
+#! /bin/bash
+
+git_toplevel="$(git rev-parse --show-toplevel)"
```
Comment (nitsanavni) [inline session-notes/cleanup-notes/doit:3]: Learn something new every day

File: session-notes/session-notes-2022-10-13.md [hunk 8]
```diff
@@ -1,3 +1,4 @@
+# Session Date: 2022-10-13
```
Comment (nitsanavni) [inline session-notes/session-notes-2022-10-13.md:1]: 🏵️

File: session-notes/cleanup-notes/cleanup_session_notes.prompt [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+  - action: if there is a 1st or 2nd level header that matches "Inactive Co-Authors", delete the whole section (i.e. the header through, but not including, the next header)
+  - return: none.  But assure that the argument has changes if they were made
+
+### Sub-task: Create subroutine 'normalize_coauthor_heading'
+  - argument: contents
+  - action: Any header that contains "Co-Author", whether or not it contains other text, should be rewritten to be a 2nd-level header with only "Co-Authors"
+  - return: none.  But assure that the argument has changes if they were made
+
+### Sub-task: Create subroutine 'cleanup_file'
+  - argument: filename
```
Comment (nitsanavni) [inline session-notes/cleanup-notes/cleanup_session_notes.prompt:56]: English as a programming language 👏
I wonder how much **less** specific can you go, letting the LLM infer and be creative, and still **reliably** generate a valuable working script

File: session-notes/Co-Authored-By_count_after_cleanup [hunk 10]
```diff
@@ -0,0 +1,27 @@
+  43 ## Co-Authors
```
Comment (nitsanavni) [inline session-notes/Co-Authored-By_count_after_cleanup:1]: Are those files necessary?

**Review by nitsanavni — COMMENTED** [review]
Love it!

**Review by nitsanavni — COMMENTED** [review]
.

File: session-notes/cleanup-notes/cleanup_session_notes.prompt (hunk 9 above)
Comment (MichaelRWolf) [reply session-notes/cleanup-notes/cleanup_session_notes.prompt:56]: I like the way you're thinking.

Let's explore it.

File: session-notes/Co-Authored-By_count_after_cleanup (hunk 10 above)
Comment (MichaelRWolf) [reply session-notes/Co-Authored-By_count_after_cleanup:1]: No, but I wanted some way to show results.

Suggestions for a better place to archive the quick-and-dirty data?

**Pull request merged**